lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -233,9 +257,358 @@ SESERegionBuilder::processAcyclicRegionExcludingEnd(SILBasicBlock *startBB,
   }
 }
 
+/// For loops in the SIL Function with multiple exit block successors,
+/// transforms the loop by introducing a common exit switch block that demuxes
+/// to the appropriate exit block. Updates the Dominanceinfo, PostDominanceinfo,
+/// and LoopInfo as well.
+bool SESERegionBuilder::ensureSingleExitFromLoops() {
+  // TODO: update changed appropriately.
+  // Visit the loop nest hierarchy bottom up.
+  bool changed = false;
+  llvm::SmallVector<std::pair<SILLoop *, bool>, 16> workList;
+  for (auto *L : LI.getTopLevelLoops())
+    workList.push_back({L, L->empty()});
+  while (workList.size() != 0) {
+    SILLoop *loop;
+    bool visited;
+    std::tie(loop, visited) = workList.pop_back_val();
+
+    if (!visited) {
+      workList.push_back({loop, true});
+      for (auto *Subloop : loop->getSubLoopRange()) {
+        workList.push_back({Subloop, Subloop->empty()});
+      }
+      continue;
+    }
+    changed |= ensureSingleExitFromLoop(loop);
+  }
+  if (changed) {
+    splitAllCondBrCriticalEdgesWithNonTrivialArgs(*F, nullptr, &LI);
+
+    DI.recalculate(*F);
+    PDI.recalculate(*F);
+  }
+  return changed;
+}
+
+namespace {
+
+/// Appends the given arguments to the given edge. Deletes the old TermInst
+/// and returns a new TermInst with the appropriate number of args.
+TermInst* appendArguments(TermInst *termInst, SILBasicBlock *target,
```

Comment:
The local convention is `TermInst *appendArguments`

Reply:
Done.